incubator/hnc/internal/validators/object.go

```diff
@@ -244,16 +246,19 @@ func (o *Object) handleInherited(op admissionv1beta1.Operation, newSource, oldSo
 	return deny(metav1.StatusReasonInternalError, "unknown operation: "+string(op))
 }
 
-// isDeletingNS returns true if the namespace of the object is already being deleted, based on the
-// in-memory forest.
-func (o *Object) isDeletingNS(inst *unstructured.Unstructured) bool {
-	o.Forest.Lock()
-	defer o.Forest.Unlock()
-	ns := o.Forest.Get(inst.GetNamespace())
-	if ns == nil {
+// isDeletingNS returns true if the namespace of the object is already being deleted
+func (o *Object) isDeletingNS(ctx context.Context, inst *unstructured.Unstructured) bool {
+
+	ns, err := o.K8sClient.CoreV1().Namespaces().Get(ctx, inst.GetNamespace(), metav1.GetOptions{})
+	if err != nil && strings.HasSuffix(err.Error(), "not found") {
```

Comment:
Use `IsNotFound`, e.g.: https://github.com/kubernetes-sigs/multi-tenancy/blob/2f6efd498c7eb6b39cbaf3a59b6c60879153c352/incubator/hnc/internal/reconcilers/hierarchy_config.go#L107

Reply:
Updated in latest commit